cocos/platform/CCImage.cpp

```diff
@@ -2165,10 +2170,10 @@ bool Image::initWithRawData(const unsigned char * data, ssize_t dataLen, int wid
         _height   = height;
         _width    = width;
         _hasPremultipliedAlpha = preMulti;
-        _renderFormat = Texture2D::PixelFormat::RGBA8888;
+        _renderFormat = renderFormat;
 
-        // only RGBA8888 supported
-        int bytesPerComponent = 4;
+        // only RGBA8888, BGRA8888 and RGB888 supported
+        int bytesPerComponent = (_renderFormat == Texture2D::PixelFormat::RGB888) ? 3 : 4;
```

Comment:
If you'd like to give me a list of all the `pixelFormats` you'd like to support in this PR, I'd be happy to add them all in.

Another option is to add in the protection that this function lacked.  We could simply throw an error if the pixel format is not supported.

Or we could do both. :smile:   Your call.